pkg/daemon/ceph/osd/volume.go

```diff
@@ -544,6 +533,46 @@ func (a *OsdAgent) useRawMode(context *clusterd.Context, pvcBacked bool) (bool,
 	return useRawMode, nil
 }
 
+func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceOsdMapping, allowRawMode bool) error {
+	// it's a little strange to split this into parts, looping here and in the init functions, but
+	// the LVM mode init requires the ability to loop over all the devices looking for metadata.
+	rawDevices := &DeviceOsdMapping{
+		Entries: map[string]*DeviceOsdIDEntry{},
+	}
+	lvmDevices := &DeviceOsdMapping{
+		Entries: map[string]*DeviceOsdIDEntry{},
+	}
+
+	for name, device := range devices.Entries {
+		// Even if we can use raw mode, do NOT use raw mode on disks. Ceph bluestore disks can
+		// sometimes appear as though they have "phantom" Atari (AHDI) partitions created on them
+		// when they don't in reality. This is due to a series of bugs in the Linux kernel when it
+		// is built with Atari support enabled. This behavior does not appear for raw mode OSDs on
+		// partitions, and we need the raw mode to create partition-based OSDs. We cannot merely
+		// skip creating OSDs on "phantom" partitions due to a bug in `ceph-volume raw inventory`
+		// which reports only the phantom partitions (and malformed OSD info) when they exist and
+		// ignores the original (correct) OSDs created on the raw disk.
+		// See: https://github.com/rook/rook/issues/7940
+		if device.DeviceInfo.Type != sys.DiskType && allowRawMode {
```

Comment:
So partitions will still use raw mode, right? How about encouraging people to put a large partition on their device and then let Rook discover it? So that users can still benefit from raw mode.

Reply:
Would you be able to write up that blurb? I think you have a better understanding of the benefits of raw mode.